### ISSUE TITLE
Suport for libimobiledevice

### ISSUE THREAD

**Issue by steeve** [bazelbuild/rules_apple#363]
So I made build targets for `libimobiledevice`, in order to use from bazel.
Are you guys interested in a PR ? That would enable `bazel run`ing on devices too.

**Comment by steeve** [comment]
Sample targets
```python
        "_idevicedebugserverproxy": attr.label(
            default = "@com_github_libimobiledevice_libimobiledevice//:idevicedebugserverproxy",
            cfg = "host",
            executable = True,
        ),
        "_ideviceinfo": attr.label(
            default = "@com_github_libimobiledevice_libimobiledevice//:ideviceinfo",
            cfg = "host",
            executable = True,
        ),
        "_ideviceimagemounter": attr.label(
            default = "@com_github_libimobiledevice_libimobiledevice//:ideviceimagemounter",
            cfg = "host",
            executable = True,
        ),
        "_ideviceprovision": attr.label(
            default = "@com_github_libimobiledevice_libimobiledevice//:ideviceprovision",
            cfg = "host",
            executable = True,
        ),
        "_ideviceinstaller": attr.label(
            default = "@com_github_libimobiledevice_libimobiledevice//:ideviceinstaller",
            cfg = "host",
            executable = True,
        ),
```

**Comment by keith** [comment]
I think this would be nice! I'm not exactly sure what the UX of this would be but bazel supports this for android so this would be nice to see work for iOS as well

**Comment by steeve** [comment]
I'll open a PR adding the targets as `third_party` and we'll get the discussion going from there.
I also have a companion script that runs the apps inside a lldb debugging session, enabling stdout redirection and so on but it wasn't stable.

**Comment by keith** [comment]
FWIW I don't speak for this repo tho so it's up to others if they would accept this

**Comment by rahul-malik** [comment]
This would be awesome, please put up a PR with everything

**Comment by steeve** [comment]
@keith yes I saw, but perhaps other people will feel on carrying the torch!

**Comment by steeve** [comment]
I've opened #364.

**Comment by sergiocampama** [comment]
sorry for the late reply.

I think this support brings in too many dependencies for the benefit we get. What do you all think on creating a runner interface similar to what the tests already do? This way, you could provide many different runners, think of a contrib repo with different functionalities. This way rules_apple would limit itself to the bundling of Apple targets.

This came up in a similar fashion in #225 too.

**Comment by sergiocampama** [comment]
Right, I was thinking something similar to what @ob did in https://github.com/linkedin/bluepill/pull/327, this repo can be added as you mention through a WORKSPACE dependency, and then added to `ios_application` through a new `runner` attribute.

**Comment by steeve** [comment]
That looks alright! Is there any API you have in mind for that runner to work? Like, that the IPA as argv[1] or something? If you look at #364, which has a functionnal `bazel run` poc, it needs a few things (like the mobileprovision file)

**Comment by sergiocampama** [comment]
It'll be similar to the AppleTestRunnerInfo, in which there's a template file that accepts a substitution to the bundle or archive (depending on whether tree artifacts outputs have been enabled)

**Comment by sergiocampama** [comment]
Sadly, `bazel run` can't be configured to add environment variables or arguments to a script, so the substitution is the only possible path forward for now

**Comment by steeve** [comment]
@sergiocampama it's absolutely fine to use substitution
